python/ray/_private/worker.py

```diff
@@ -830,8 +830,8 @@ def main_loop(self):
         """The main loop a worker runs to receive and execute tasks."""
 
         def sigterm_handler(signum, frame):
-            shutdown(True)
             sys.exit(1)
+            shutdown(True)
```

Comment:
This is the main fix. exit(1) will cause the worker to shutdown actor execution, and then shutdown will happen properly after.